docs/gym-skill.md

```diff
@@ -91,4 +91,14 @@ This diagram shows the communication between the AEA and the gym environment
         deactivate Environment
 </div>
 
+## Skill Architecture
+
+The skill consists of two core components: `GymHandler` and `GymTask`.
+
+In the `setup` method of the `GymHandler` the `GymTask` is initialized, as well as its `setup` and `execute` methods called. The handler, which is registered agains the `GymMessage.protocol_id` then filters for messages of that protocol with the performative `GymMessage.Performative.PERCEPT`. These messages are passed to the `proxy_env_queue` of the task.
+
+The `GymTask` is responsible for training the RL agent. In particular, `MyRLAgent` is initialized and trained against `ProxyEnv`. The `ProxyEnv` instantiates a `gym.Env` class and therefore implements its API. This means, the proxy environment is compatible with any `gym` compatible RL agent. However, unlike other environments it only acts as a proxy and does not actually implement an environment of its own. It allows for the decoupling of the process environment of the `gym.env` from the process environment of the RL agent. The actual `gym.env` against which the agent is trained is wrapped by the `gym` connection. The proxy environment and gym connection communicate via a protocol, the `gym` protocol. Note, it would trivially be possible to implement the `gym` environment in another AEA; this way one AEA could provide `gym` environments as a service. Naturally, the overhead created by the introduction of the extra layers causes a higher latency when training the RL agent.
```

Comment:
- This means the proxy (without a comma)
- However, unlike other environments, (with a comma)

Reply:
- you can drop the `actually` from the sentence if you want.